src/Markup/Avalonia.Markup.Xaml/XamlIl/Runtime/XamlIlRuntimeHelpers.cs

```diff
@@ -80,6 +82,8 @@ IEnumerable<object> GetParents()
                     return this;
                 if (serviceType == typeof(IAvaloniaXamlIlControlTemplateProvider))
                     return this;
+                if (serviceType == typeof(IRuntimePlatform))
+                    return _runtimePlatform;
```

Comment:
This service is rarely used, but DeferredParentServiceProvider is often created (per each deferred resource).
So it would make sense to call GetService only when it's actually needed in GetService.

It's the opposite in the global service provider, which is created once, so it's fine to pre-save runtime platform there.